src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -191,13 +206,13 @@ pub mod tests {
         let cm = ConcurrencyManager::new(TimeStamp::zero());
         let mut txn = MvccTxn::new(start_ts, cm);
         let snapshot = engine.snapshot(Default::default()).unwrap();
-        let mut reader = SnapshotReader::new_with_ctx(version, snapshot, &ctx);
+        let mut reader = SnapshotReader::new_with_ctx(start_ts, snapshot, &ctx);
```

Comment:
Emm... The `start_ts` here is just the ts of the flashback command? Do we guarantee that it equals to the start ts of the locks? If not, then that doesn't sound correct...

Reply:
It's not guaranteed, so it looks like we'll have to rollback the lock one by one according to its own `lock.ts` rather than the flashback `start_ts`. @HuSharp

Reply:
Yes...

Reply:
Or we can set `reader.start_ts` to `lock.ts` each time before we call `rollback_lock`.